plugin_tests/cli_results_test.py

```diff
@@ -0,0 +1,167 @@
+                        while True:
+                            data = fptr.read(chunkSize)
+                            if not data:
+                                break
+                            sha.update(data)
+                    try:
+                        self.assertEqual(options['hash'], sha.hexdigest())
+                    except Exception:
+                        sys.stderr.write('File hash mismatch: %s\n' % outpath)
+                        raise
+                if 'contains' in options:
+                    data = open(outpath, 'rb').read(chunkSize)
+                    for entry in contains:
+                        self.assertIn(entry, data)
+            for entry in contains:
+                self.assertIn(entry, stdout)
+            for entry in excludes:
+                self.assertNotIn(entry, stdout)
+        except Exception:
+            sys.stderr.write('CMD (cwd %s):\n%r\n' % (cwd, cmd))
+            sys.stderr.write('STDOUT:\n%s\n' % stdout.rstrip())
+            raise
+        finally:
+            shutil.rmtree(tmppath)
+        return stdout
+
+    def testListCLI(self):
+        """
+        Test that we can list clis, get help for each, and get the xml spec for
+        each.
+        """
+        from girder.plugins.slicer_cli_web import rest_slicer_cli
+        from girder.api.rest import Resource
+
+        restResource = Resource()
+        cli_args = ('--list_cli', )
+        cli_list = self._runTest(cli_args, contains=['"NucleiDetection"'])
+        cli_list = json.loads(cli_list)
+        self.assertIn('NucleiDetection', cli_list)
+        for cli in cli_list:
+            cli_args = (cli, '--help')
+            # Each cli's help must mention usage, its own name, and that you
+            # can output xml
+            self._runTest(cli_args, contains=['usage:', cli, '--xml'])
+            cli_args = (cli, '--xml')
+            # The xml output needs to have a tile and an executable tag
+            xml = self._runTest(cli_args, contains=[
+                '<executable>', '<title>', '</executable>', '</title>'])
+            if '<' in xml:
+                xml = xml[xml.index('<'):]
+            try:
+                rest_slicer_cli.genHandlerToRunDockerCLI(
+                    'dockerimage', os.environ.get('CLI_CWD'), xml, restResource)
+            except Exception:
+                sys.stderr.write('Failed in generating endpoints for %s' % cli)
+                raise
+
+    def testNucleiDetectionDefaults(self):
+        cli_args = (
+            'NucleiDetection',
+            os.path.join(TEST_DATA_DIR, 'Easy1.png'),
+            'tmp_1.anot',
+        )
+        cli_kwargs = {
+            'analysis_roi': '-1.0, -1.0, -1.0, -1.0',
+        }
+        self._runTest(cli_args, cli_kwargs, outputs={
+            'tmp_1.anot': {
+                'contains': ['elements'],
+                # 'hash': '02b240586412c87ad5cbf349b7c22f80f1df31eef54ed8ee4ad1fd3624a89fa2',
```

Comment:
Requiring the sha hash to match could be too rigid for comparing image analysis results.

But i guess we can extend this later.